pallets/subtensor/src/uids.rs

```diff
@@ -9,6 +9,28 @@ impl<T: Config> Pallet<T> {
         SubnetworkN::<T>::get(netuid)
     }
 
+    /// Returns a callback that sets the element at the given position to zero, doing nothing if the
+    /// position is out of bounds
+    fn clear_element_at<N>(position: u16) -> impl Fn(&mut Vec<N>)
```

Comment:
What is the advantage of using this over `take` or `remove` ?

Reply:
the vector is indexed by uid so we don't want to remove an element from it - just set it to zero (and check for bounds)